src/app/groups/[groupId]/expenses/[expenseId]/edit/page.tsx

```diff
@@ -36,7 +37,13 @@ export default async function EditExpensePage({
 
   async function deleteExpenseAction() {
     'use server'
-    await deleteExpense(expenseId)
+    await deleteExpense(groupId, expenseId)
+    redirect(`/groups/${groupId}`)
+  }
+
+  async function archiveExpenseStateUpdateAction(state: boolean) {
```

Comment:
Is this stuff about archiving related to this commit "added locks to ensure only 1 user can get the group expenses created"? If no, can it be factored out of this commit / pull request and submitted separately?

Reply:
No
This feature was added here for removing a recurring transction temporarily.
For example:
I have a monthly subscription of Netflix. It was there for 6 months but we cancelled for 1 month.
It will automatically add in the recurring expense section.
If I delete the transaction, future recurring transactions will not be created. Hence Archiving a transaction will result it to be there, but won't add in the split calculations.

Reply:
Please check [this image](https://private-user-images.githubusercontent.com/52707760/311215459-19c44392-7393-4045-b62b-454f29e3f125.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI3ODAxMDksIm5iZiI6MTcxMjc3OTgwOSwicGF0aCI6Ii81MjcwNzc2MC8zMTEyMTU0NTktMTljNDQzOTItNzM5My00MDQ1LWI2MmItNDU0ZjI5ZTNmMTI1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDEwVDIwMTAwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRlOGFhODkyOWU3Y2YzZTgwZGE3YzE4MjNlYWVmZGI2MTRlM2RkMjY2MDM2YjU0OWM2YTEwYWFhMGJhYmFjNjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NR9C9_DO80OMkDkzg6frjjxQTT2dpFbJVssUq4ya8Hg) for archived expense in the fetched expenses

Reply:
I see. I recommend simplifying the pull request to remove archived expenses for now, and propose that as a separate feature. There may be disagreement over the user experience here and it will delay acceptance of the recurring expense feature. (The word "archive" to me does not connote the behavior of the feature, for example.)

Splitwise behavior is: The most recent expense is the one marked "recurring." Older expenses can be deleted without affecting the recurrence. If you delete the most recent expense, the recurrence stops.

That makes it easy to delete transactions 1...n-1 without affecting the recurrence.But I agree it is useful to have a way to delete the most recent transaction without affecting the recurrence, too.

Reply:
sure. Can do. I'll raise that feature as another PR and add an alert.
Archive seemed best to me as it just serves as a historical preserved record for reference. Any other suggestion?